cache/metadata.go

```diff
@@ -307,6 +308,29 @@ func getSize(si *metadata.StorageItem) int64 {
 	return size
 }
 
+func queueBlobSize(si *metadata.StorageItem, s int64) error {
```

Comment:
Yeah previously we were saving the size as calculated by the [`Size` method](https://github.com/moby/buildkit/blob/6f83882a0b098034817f37858c32d45133c5703c/cache/refs.go#L138) in a separate metadata field, but that is the actual disk space taken up (including unpacked snapshots). `BlobSize` on the other hand is just the size of the packed blob as specified in the oci descriptor (whether or not the blob actually exists on disk). I had to add it because the [`contentutil.Copy`](https://github.com/sipsma/buildkit/blob/a7499cd28837af1654f3782e7866d5103371395a/cache/refs.go#L359) func was returning errors about mismatched hashes if `Size` wasn't specified in the descriptor.